apis/components/v1alpha1/codeflare_types.go

```diff
@@ -30,6 +30,11 @@ const (
 
 // CodeFlareCommonStatus defines the shared observed state of CodeFlare
 type CodeFlareCommonStatus struct {
+	// +patchMergeKey=name
+	// +patchStrategy=merge
+	// +listType=map
+	// +listMapKey=name
+	Releases []common.ComponentReleaseStatus `json:"releases,omitempty"`
```

Comment:
Maybe create a `ComponentReleaseStatus` and embed it here so you don't have to repeat the annotation in each component